common-react/@dbeaver/ui-kit/src/Select/SelectField.tsx

```diff
@@ -0,0 +1,143 @@
+import { useState } from 'react';
+import { SelectProvider, Select, SelectPopover, SelectItem, SelectLabel } from './Select.js';
+import './SelectField.css';
+
+export interface SelectOption<T = string> {
+  value: T;
+  label: string;
+  disabled?: boolean;
+}
+
+type PropertyPath = string;
+type PropertyGetter<ItemType, ValueType> = PropertyPath | ((item: ItemType) => ValueType);
```

Comment:
we need to use typesafe PropertyPath calculated based on ValueType and ItemType:
You need to filter fields in ItemType so only items that satisfy ValueType then we need to extract keys of this fields as fields that can be used